### PR TITLE
[klippa] use serializer framework

### PR DESCRIPTION
**Pull request by qxliu76**

### CODE REVIEW

File: klippa/src/name.rs [hunk 1]
```diff
@@ -50,62 +56,70 @@ impl Subset for Name<'_> {
...
-            let str_data = name_data
-                .get(str_start..str_start + str_len as usize)
-                .ok_or(SubsetTableError(Name::TAG))?;
-            out.extend_from_slice(str_data);
+        let record = name_records[*idx];
+        let offset = record.string_offset().to_u32() as usize;
 
-            string_offset += str_len;
-        }
+        let offset_pos = len + 10;
```
Comment (cmyr) [inline klippa/src/name.rs:97]: is the `10` here the length of a NameRecord?

File: klippa/src/serialize.rs [hunk 2]
```diff
@@ -198,6 +198,35 @@ impl Serializer {
         Ok(ret)
     }
 
+    //copy from a single Scalar type
```
Comment (cmyr) [inline klippa/src/serialize.rs:201]: if it's a doc comment you can use three `///`

**Review by cmyr — APPROVED** [review]
looks reasonable, a few things inline :)

File: klippa/src/name.rs (hunk 1 above)
Comment (qxliu76) [reply klippa/src/name.rs:98]: added comment:)

File: klippa/src/lib.rs [hunk 3]
```diff
@@ -523,65 +525,126 @@ pub fn subset_font(font: &FontRef, plan: &Plan) -> Result<Vec<u8>, SubsetError>
...
+        s.end_serialize();
+        return ret;
+    }
+
+    // ran out of room, reallocate more bytes
+    let buf_size = s.allocated() * 2 + 16;
+    if buf_size > (table_len as usize) * 256 {
+        return ret;
+    }
+    s.reset_size(buf_size);
```
[the remark this answers is not in the document]
Comment (qxliu76) [reply klippa/src/lib.rs:592]: I'd like to keep serializer API simpler.:)

**Review by garretrieger — APPROVED** [review]

**Pull request merged**